### ISSUE TITLE
Query "where" unexpected behavior.

### ISSUE THREAD

**Issue by goliatone** [balderdashy/waterline-criteria#24]
[Originally posted at waterline repo, moved here. See original issue [here]]

I have an entity with a **trackingNumber** field of type `string`. The field is unique.

Querying by trackingNumber I expect to get one and only one record. Instead I get two. If you look at the **trackingNumber** value on each record, you can see that they are very similar.
- **333230389361289949034304032**_696_
- **333230389361289949034304032**_726_

```json
// 20150707112846
// http://localhost:3030/api/packages?where={"trackingNumber":"420100389361289949054104032726"}

[
  {
    "uuid": "e9201f96-ec85-4faf-8a00-977f6a849222",
    "trackingNumber": "333230389361289949034304032696",
    "status": "delivered",
    "id": 2,
    "createdAt": "2015-07-06T15:14:51.446Z",
    "updatedAt": "2015-07-06T16:49:48.187Z"
  },
  {
    "trackingNumber": "333230389361289949034304032726",
    "status": "received",
    "uuid": "e9201f96-ec85-4faf-8a00-977f6a849498",
    "createdAt": "2015-07-07T15:27:31.344Z",
    "updatedAt": "2015-07-07T15:27:31.429Z",
    "id": 6
  }
]
```

If I query by **uuid** works as expected. Note that the values are also pretty similar:
- **e9201f96-ec85-4faf-8a00-977f6a849**_222_
- **e9201f96-ec85-4faf-8a00-977f6a849**_498_

```json
/ 20150707130619
// http://localhost:3030/api/packages?where={"uuid":"e9201f96-ec85-4faf-8a00-977f6a849222"}

[
  {
    "uuid": "e9201f96-ec85-4faf-8a00-977f6a849222",
    "trackingNumber": "333230389361289949034304032696",
    "status": "delivered",
    "id": 2,
    "createdAt": "2015-07-06T15:14:51.446Z",
    "updatedAt": "2015-07-06T16:49:48.187Z"
  }
]
```

```
"waterline": "^0.10.24",
"sails-disk": "^0.10.8",
```
[here]: https://github.com/balderdashy/waterline/issues/1082

**Comment by dmarcelino** [comment]
For reference, @particlebanana said in https://github.com/balderdashy/waterline/issues/1082#issuecomment-119694267:

> Looks like a bug in [waterline-criteria](https://github.com/balderdashy/waterline-criteria) where it's seeing a number that's a string and trying to do weird coercion things to it. Shouldn't be a problem in any other databases besides disk and memory.

**Comment by mikermcneil** [comment]
Thanks @dmarcelino and @goliatone!

As @particlebanana said, looks to me like what's going on here is an integer type overflow (which shouldn't happen because the input is a string).  I suspect [`isNumbery`](https://github.com/balderdashy/waterline-criteria/blob/master/lib/filters/where.js#L247).

**Comment by sailsbot** [comment]
Thanks for posting, @goliatone.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by devinivy** [comment]
Active bug– reopening.
